src/libraries/System.DirectoryServices/tests/System/DirectoryServices/DirectoryServicesTests.cs

```diff
@@ -14,6 +15,13 @@ public partial class DirectoryServicesTests
         internal static bool IsLdapConfigurationExist => LdapConfiguration.Configuration != null;
         internal static bool IsActiveDirectoryServer => IsLdapConfigurationExist && LdapConfiguration.Configuration.IsActiveDirectoryServer;
 
+        [Fact]
+        public void TestGetAllTypes()
+        {
+            Type[] allTypes = typeof(DirectoryEntry).Assembly.GetTypes();
```

Comment:
I wonder whether it would be possible (later) to write a test that did this for each of our implementation assemblies, without actually pasting in the test into every one.

Something like - a common file that was included in all test assemblies, with a test that enumerated all loaded assemblies and did GetTypes on them.

Reply:
I opened https://github.com/dotnet/runtime/issues/68341 to track trying to do something like this.